packages/jetbrains-plugin/src/main/kotlin/com/mongodb/jbplugin/editor/EditorToolbarDecorator.kt

```diff
@@ -169,9 +139,17 @@ class EditorToolbarDecorator(
         }
 
         val javaPsiFile = psiFile as PsiJavaFile
-        return arrayOf(
-            this::isUsingTheJavaDriver,
-        ).any { it(javaPsiFile) }
+        if (isUsingSpringDataMongoDb(javaPsiFile)) {
+            toolbar.showDatabaseSelector()
```

Comment:
These show and hide actions feels like a sideeffect in this method, would they be better suited when done in the `ensureToolbarIsVisibleIfNecessary`, right after we assign the toolbar to editor.permanentHeaderComponent?

Reply:
The issue here is that showing the database selector depends on the driver the developer is using. We have 3 different scenarios:

* Not using MongoDB -> do not show the toolbar
* Using the Java Driver -> show the toolbar with the data source selector, but hide the database selector
* Using Spring Criteria -> show the toolbar with both the data source and database selectors

Moving this logic upwards would mean that we need to either reparse the file to check which driver is using, or returning some value (an enum maybe?) that we would use to decide what to show.

Reply:
yea I was kinda implying that but if it feels a little too much then we can also just rename the method to signify that "it checks wether its a java file and perform a side-effect"

Reply:
Changed to `isToolbarSetUpForCurrentFile` which seems more accurate.